### ISSUE TITLE
No VariableCreate event when dragging a block with a variable out of the flyout

### ISSUE THREAD

**Issue by rachel-fenichel** [google/blockly#1395]
### Problem statement

- [x] Bug report
- [ ] Feature request

Dragging a block with a variable out of the flyout fires a create event for the block but not for the variable, even though the variable is created on the workspace.

This is for a flyout block with a variable that is not already in the workspace.  e.g. 
![image](https://user-images.githubusercontent.com/13686399/32077499-7a48b856-ba58-11e7-9c18-b86bbe26a758.png)

### Expected Behavior

Variable creation events are fired when variables are created.

### Actual Behavior

Only a block creation event is fired.

### Steps to Reproduce

Enable "log events" on the playground.
Drag out a list block with a variable.

### Additional Information
```
Blockly.Flyout.prototype.createBlock = function(originalBlock) {
  var newBlock = null;
  Blockly.Events.disable();
  this.targetWorkspace_.setResizesEnabled(false);
  try {
    newBlock = this.placeNewBlock_(originalBlock);
    // Close the flyout.
    Blockly.hideChaff();
  } finally {
    Blockly.Events.enable();
  }

  if (Blockly.Events.isEnabled()) {
    Blockly.Events.setGroup(true);
    Blockly.Events.fire(new Blockly.Events.Create(newBlock));
  }
```

Blockly.Flyout.prototype.createBlock turns off events while placing the new block, then fires a single Create event that includes all of the blocks in the new stack.  It isn't aware of variables.  The variable create event is not fired from inside placeNewBlock because events are disabled.

I don't know why events are disabled here, instead of just grouped.

**Comment by rachel-fenichel** [comment]
Events are disabled instead of just grouped because each block would serialize the entire stack below it, meaning that there would be a lot of duplicate information and a lot of unnecessary serialization cost.

The fix is:
- Before creating the new block(s), get a list of all variables on the workspace.
- After creating the new block(s) get a list of all variables on the workspace.
- Reenable events.
- Start the event group.
- Fire the block create event.
- Compare the lengths of the two variable lists.
  - If they are the same lengths:
    - End group and return.
  - If they are different lengths:
    - Find all variables that exist in the second list and not in the first list.
    - Fire one event per created variable.
    - End group and return.

**Comment by rachel-fenichel** [comment]
Fixed by #1477
